surfsense_backend/app/utils/connector_service.py

```
@@ -479,4 +479,83 @@ async def search_extension(self, user_query: str, user_id: int, search_space_id:
+            # Fix for UI
+            youtube_chunks[i]['document']['id'] = self.source_id_counter
+            
+            # Extract document metadata
+            document = chunk.get('document', {})
+            metadata = document.get('metadata', {})
+
+            # Extract YouTube-specific metadata
+            video_title = metadata.get('video_title', 'Untitled Video')
+            video_id = metadata.get('video_id', '')
+            channel_name = metadata.get('channel_name', '')
+            published_date = metadata.get('published_date', '')
```

Ruff (0.8.2)

Local variable `published_date` is assigned to but never used

Remove assignment to unused variable `published_date`

(F841)